content/articles/shell-script-to-demonstrate-the-se-of-shell-function-library/index.md

```diff
@@ -1,28 +1,71 @@
+- how to create a shell function.
+- how to use functions from the library and then demonstrate how to use bash script as a function library.
 
+### How to create a function
+As described in the introduction, a function is a block of code that carries out a certain activity. A function can be Called and reused. In shell scripting, functions are analogous to other programming languages' subroutines, procedures, and functions.
+
+Simply enter the following code to create a function:
+
+```bash
+function_name () { 
+   list of commands
+}
+```
+
+Your function's name is function_name, and you'll call it by that name throughout your scripts. The name of the function must be preceded by parentheses and a series of instructions surrounded by braces.
+
+Below is an example to show the use of a function:
+
+```bash
+Hello () {
+   echo "Hello World"
+}
+
+Hello
+``` 
+Save the script above in your Linux system as `test.sh` and then run it in the terminal by writing `./test.sh`
+
+The script above will display the following output.
+
+```bash
+Hello World
+```
+In C/C++, a Shell Function Library is similar to a header file, whereas, in Python, it is similar to a module. The current script must be aware of the function's location. To be able to use the functions, we must additionally add the file's location to the Environment variables, or run the script in the terminal to temporarily load the function library into the current shell. Shell Script Functions are similar to regular shell scripts in that they only define functions.
 
 ### Why use Function Library
+
 The most important reason for having a function library is for reusability. We require functions in a function library for the following reasons.
 
-1. Sometimes we require logic to address looping logic. For example, in Oracle Applications, whenever a concurrent request is sent, we usually click on the Refresh button to retrieve the most recent data about the concurrent program's status. Or, we may create a function and place it in the function library so it can be used in other scripts.
-2. We could need a collection of functions to deal with an excel file, a CSV file, an XML file, or a properties file at times. This is the situation when we wish to work with and would need across several test automation scripts, so once the functions are developed, they can be reused wherever they are needed. We may need to interact with these files for a variety of reasons, including test data iterations, spreadsheet verification, and so on.
+1. Sometimes we require logic to address looping logic. [To know more about looping logic click here](https://docs.oracle.com/cd/E13214_01/wli/docs102/bpguide/bpguideWhile.html) For example, in Oracle Applications, whenever a concurrent request is sent, we usually click on the Refresh button to retrieve the most recent data about the concurrent program's status. Or, we may create a function and place it in the function library so it can be used in other scripts.

```

Comment:
@nicasious  replace this  [To know more about looping logic click here] with [looping logic]